app/models/application_record.rb

```diff
@@ -77,32 +77,47 @@ def pending?
     !reviewed
   end
 
-  def self.gender_eeo_data(start_date, end_date, department_ids)
-    gender_records = between(start_date, end_date).with_gender
-                                                  .in_department(department_ids)
+  def self.combined_eeo_data(records)
+    combined_records = records.with_gender.with_ethnicity
+    sub_hash = {}
+    all_ethnicities = ETHNICITY_OPTIONS | combined_records.pluck(:ethnicity)
+    all_genders = GENDER_OPTIONS | combined_records.pluck(:gender)
+    all_genders.map do |gender|
+      sub_array = []
+      all_ethnicities.map do |ethnicity|
+        records = combined_records.where(ethnicity: ethnicity,
+                                         gender: gender)
+        sub_array << [ethnicity, records.count, records.interview_count]
+        sub_hash.store(:"#{gender}", sub_array)
```

Comment:
I think she wants it to be a symbol (note: this won't work if gender is an integer, but I don't think it is)

```ruby
sub_hash[gender.to_sym] = sub_array
``